endlesscodex/src/component/itemlayout/item-layout.component.tsx

```diff
@@ -12,27 +12,40 @@ import {
 
 interface ItemLayoutProps {
   children: ReactNode;
-  /* eslint-disable react/require-default-props */
+  // eslint-disable-next-line react/require-default-props
   className?: string;
+  // eslint-disable-next-line react/require-default-props
+  disableAnimation?: boolean; // Nouvelle prop
 }
 
-function ItemLayoutComponent({ children, className }: ItemLayoutProps) {
+function ItemLayoutComponent({
+  children,
+  className,
+  disableAnimation,
+}: ItemLayoutProps) {
   const dispatch = useDispatch();
   const isVisible = useSelector((state: RootState) => state.item.visible);
 
   useEffect(() => {
-    dispatch(showItem());
+    if (!disableAnimation) {
+      dispatch(showItem());
+    }
 
     return () => {
-      dispatch(hideItem());
+      if (!disableAnimation) {
+        dispatch(hideItem());
+      }
     };
-  }, [dispatch]);
+  }, [dispatch, disableAnimation]);
+
+  // eslint-disable-next-line no-nested-ternary
+  const scaleValue = disableAnimation ? 1 : isVisible ? 1 : 0;
```

Comment:
je pense que la ligne ci-dessous fait le même résultat et te permet de retirer le // eslint-disable-next-line no-nested-ternary
```suggestion
  const scaleValue = disableAnimation || isVisible ? 1 : 0;
```